pkg/iptableswrapper/mocks/iptables_maps.go

```diff
@@ -41,10 +41,16 @@ func (ipt *MockIptables) Exists(table, chainName string, rulespec ...string) (bo
 }
 
 func (ipt *MockIptables) Insert(table, chain string, pos int, rulespec ...string) error {
+	// We only have -N chainName in the map for chains we manually create
+	if !strings.HasPrefix(chain, "AWS-SNAT-CHAIN") && !strings.HasPrefix(chain, "AWS-CONNMARK-CHAIN") {
+		pos = pos - 1
+	}
 	if ipt.DataplaneState[table] == nil {
 		ipt.DataplaneState[table] = map[string][][]string{}
 	}
-	ipt.DataplaneState[table][chain] = append(ipt.DataplaneState[table][chain], rulespec)
+	ipt.DataplaneState[table][chain] = append(ipt.DataplaneState[table][chain], []string{})
+	copy(ipt.DataplaneState[table][chain][pos+1:], ipt.DataplaneState[table][chain][pos:])
+	ipt.DataplaneState[table][chain][pos] = rulespec
```

Comment:
The `copy` is a bit confusing to read. A simple `append` should work here. Can we follow the logic of something like https://stackoverflow.com/questions/46128016/insert-a-value-in-a-slice-at-a-given-index